File: CleanCodeJava/app/src/main/java/io/chuumong/cleancodejava/di/main/MainComponent.java
package io.chuumong.cleancodejava.di.main;

import dagger.Component;
import dagger.Subcomponent;
import io.chuumong.cleancodejava.di.ActivityScope;
import io.chuumong.cleancodejava.di.ApiModule;
import io.chuumong.cleancodejava.ui.activity.MainActivity;

/**
 * Created by JongHunLee on 2017-02-22.
 */

@ActivityScope
@Subcomponent(modules = MainModule.class)
public interface MainComponent {

    void inject(MainActivity activity);
}

File: CleanCodeKotlin/app/src/main/java/io/chuumong/cleancodekotlin/di/main/MainComponent.kt
package io.chuumong.cleancodekotlin.di.main

import dagger.Subcomponent
import io.chuumong.cleancodekotlin.ui.activity.MainActivity
import io.chuumong.cleancodekotlin.di.ActivityScope

/**
 * Created by JongHunLee on 2017-02-23.
 */

@ActivityScope
@Subcomponent(modules = arrayOf(MainModule::class))
interface MainComponent {
    fun inject(activity: MainActivity)
}